### PR TITLE
add cpu random Generator

### PR DESCRIPTION
**Pull request by yaoxuefeng6**
### PR types

Others
### PR changes

Others
### Describe

bak

### CODE REVIEW

**Review by jzhang533 — DISMISSED** [review]
lgtm
this is a breaking change, the related OP should remove `ctx.Attr<int>("seed")` accordingly.
@yaoxuefeng6 will have instructions on how to fix the related OP.
@XiaoguangHu01 , @willthefrog , please have a look.

File: paddle/fluid/operators/randint_op.cc [hunk 1]
```diff
@@ -42,15 +43,20 @@ class CPURandintKernel : public framework::OpKernel<T> {
...
     std::uniform_int_distribution<T> dist(ctx.Attr<int>("low"),
                                           ctx.Attr<int>("high") - 1);
-    for (int64_t i = 0; i < size; ++i) data[i] = dist(engine);
+    if (framework::Generator::GetInstance()->is_init_py) {
+      auto& gen_engine = framework::Generator::GetInstance()->GetCPUEngine();
+      for (int64_t i = 0; i < size; ++i) data[i] = dist(gen_engine);
+    } else {
+      unsigned int seed = static_cast<unsigned int>(ctx.Attr<int>("seed"));
+      std::minstd_rand engine;
+      if (seed == 0) {
```
Comment (willthefrog) [inline paddle/fluid/operators/randint_op.cc]: what if use want to set seed to literal 0?

File: paddle/fluid/framework/generator.h [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+#include <memory>
+#include <mutex>  // NOLINT
+#include <random>
+#include <typeinfo>
+#include <utility>
+
+namespace paddle {
+namespace framework {
+
+struct GeneratorState {
```
Comment (willthefrog) [inline paddle/fluid/framework/generator.h:30]: why note simply merge this into `Generator` as it is one-to-one "has a" relationship?

File: paddle/fluid/operators/randint_op.cc (hunk 1 above)
Comment (willthefrog) [inline paddle/fluid/operators/randint_op.cc]: what if use want to set seed to literal 0?

File: paddle/fluid/framework/generator.h [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  int64_t device = -1;
+  uint64_t current_seed = 34342423252;
+  std::mt19937_64 cpu_engine;
+};
+
+struct Generator {
+  Generator() {
+    GeneratorState default_gen_state_cpu;
+    default_gen_state_cpu.device = -1;
+    default_gen_state_cpu.current_seed = 34342423252;
```
Comment (willthefrog) [inline paddle/fluid/framework/generator.h:40]: default initialize with fixed seed?

File: paddle/fluid/operators/randint_op.cc (hunk 1 above)
Comment (yaoxuefeng6) [reply paddle/fluid/operators/randint_op.cc]: Currently It is not allowed in paddle, and related doc note this to users. If it is necessary to allow this, we can fix it.

File: paddle/fluid/framework/generator.h (hunk 2 above)
Comment (yaoxuefeng6) [reply paddle/fluid/framework/generator.h:30]: To let State-related value be in this class. we can easily and clearly add things in this class later.

File: paddle/fluid/framework/generator.h (hunk 3 above)
Comment (yaoxuefeng6) [reply paddle/fluid/framework/generator.h:40]: We should set a default one. maybe random?

Comment (willthefrog) [reply paddle/fluid/framework/generator.h:40]: initialization with random seed seems more reasonable

**Review by jzhang533 — APPROVED** [review]
lgtm
will merge and start testing

**Review by luotao1 — APPROVED** [review]
LGTM for __init__.py

**Pull request merged**